cab-token-generator/java/com/google/auth/credentialaccessboundary/ClientSideCredentialAccessBoundaryFactory.java

```diff
@@ -45,41 +46,65 @@
 import com.google.auth.oauth2.StsTokenExchangeRequest;
 import com.google.auth.oauth2.StsTokenExchangeResponse;
 import com.google.common.base.Strings;
+import com.google.common.util.concurrent.SettableFuture;
 import com.google.errorprone.annotations.CanIgnoreReturnValue;
 import java.io.IOException;
+import java.time.Duration;
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import javax.annotation.Nullable;
 
 public final class ClientSideCredentialAccessBoundaryFactory {
   private final GoogleCredentials sourceCredential;
   private final transient HttpTransportFactory transportFactory;
   private final String tokenExchangeEndpoint;
   private String accessBoundarySessionKey;
   private AccessToken intermediateAccessToken;
+  private final Duration minimumTokenLifetime;
+  private final Duration refreshMargin;
+  private static final Duration DEFAULT_REFRESH_MARGIN = Duration.ofMinutes(30);
+  private static final Duration DEFAULT_MINIMUM_TOKEN_LIFETIME = Duration.ofMinutes(3);
```

Comment:
Is it possible for DEFAULT_REFRESH_MARGIN to be less than DEFAULT_MINIMUM_TOKEN_LIFETIME? Do we need any validation for user configurations?

Reply:
The default value of refresh margin is 30 minutes and the default or minimum token lifetime is 3 minutes.
Technically the users could set those values in a way that the refresh margin ends up being less than the minimum token lifetime.

Reply:
Is this something that we want to guard against? Or is this fine to allow users to do this?

Reply:
No we're not going to enforce it. The only thing we need to check is that the entered values are positive. The users might want to intentionally configure those values in a way that they only have blocking refresh, meaning refresh margin would be smaller than minimum token lifetime, and that's ok.